src/components/RegistrationForm/RegistrationForm.js

```diff
@@ -1,43 +1,101 @@
-  successRegistration,
+  handleRegistration,
+  globalErrorMessage
 }) {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [username, setUsername] = useState("");
+  const button = useRef()
+  const { handleChange, isValid, errors, resetForm } = useForm();
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    handleRegistration(email, password, username);
+  };
+  useEffect(() => {
+    if (!isOpen) {
+      resetForm();
+    }
+  }, [isOpen])
+  useEffect(() => {
+    if (!isValid) {
+      button.current.setAttribute("disabled", "");
+    }
+    if (isValid) {
+      button.current.removeAttribute("disabled", "");
+    }
+  }, [isValid])
   return (
     <PopupWithForm title={"Sign up"} isOpen={isOpen} onClose={onClose}>
-      <form className="form">
+      <form className="form" onSubmit={handleSubmit}>
         <fieldset className="form__register">
           <label className="form__label">Email</label>
           <input
             required
             name="email"
             type="email"
+            value={email}
             placeholder="Enter email"
             className="form__input"
+            onChange={(e) => {
+              setEmail(e.target.value);
+              handleChange(e);
+            }}
           />
+          <span id="input_type_password-error" className="form__span-error">
+            {errors.email ? errors.email : ""}
+          </span>
           <label className="form__label">Password</label>
           <input
             required
             name="password"
             type="password"
+            value={password}
             placeholder="Enter password"
             className="form__input"
             minLength={3}
+            onChange={(e) => {
+              setPassword(e.target.value);
+              handleChange(e);
+            }}
           />
+          <span id="input_type_password-error" className="form__span-error">
+            {errors.password ? errors.password : ""}
+          </span>
           <label className="form__label">Username</label>
           <input
             required
             name="username"
             type="text"
+            value={username}
             placeholder="Enter your username"
             className="form__input"
-            minLength={3}
+            minLength={2}
+            onChange={(e) => {
+              setUsername(e.target.value);
+              handleChange(e);
+            }}
           />
+          <span id="input_type_password-error" className="form__span-error">
+            {errors.username ? errors.username : ""}
+          </span>
         </fieldset>
-        <button onClick={successRegistration} className="form__button">
+        <span id="input_type_global-error" className="form__span-error">
+          {globalErrorMessage && globalErrorMessage}
+        </span>
+        <button
+          type="submit"
+          ref={button}
+          onClick={handleRegistration}
```

Comment:
`          onClick={handleRegistration}
`

please, delete it from the button


It doesn't work and you don't need it here


Submit handlers are added only to tags `form` with event `submit` rather than to submit buttons with event `click` because it also automatically handles `Enter` presses